modules/tinymce/src/core/test/ts/browser/textpatterns/FindBlockPatternTest.ts

```diff
@@ -0,0 +1,207 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Thunk } from '@ephox/katamari';
+import { TinyHooks, TinySelections } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+import * as Options from 'tinymce/core/api/Options';
+import * as BlockPattern from 'tinymce/core/textpatterns/core/BlockPattern';
+import * as Pattern from 'tinymce/core/textpatterns/core/Pattern';
+import { PatternSet } from 'tinymce/core/textpatterns/core/PatternTypes';
+
+const getPatternSetFor = (hook: TinyHooks.Hook<Editor>) => Thunk.cached((): PatternSet => {
+  const editor = hook.editor();
+  const rawPatterns = Options.getTextPatterns(editor);
+  const dynamicPatternsLookup = Options.getTextPatternsLookup(editor);
+  return Pattern.createPatternSet(
+    Pattern.fromRawPatterns(rawPatterns),
+    dynamicPatternsLookup
+  );
+});
+
+describe('browser.tinymce.core.textpatterns.FindBlockPatternTest', () => {
+
+  describe('no text_patterns_lookup', () => {
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      text_patterns: [
+        { start: '*', end: '*', format: 'italic' },
+        { start: '**', end: '**', format: 'bold' },
+        { start: '#', format: 'h1' },
+        { start: '##', format: 'h2' },
+        { start: '###', format: 'h3' }
+      ],
+      base_url: '/project/tinymce/js/tinymce'
+    }, []);
+
+    const getPatternSet = getPatternSetFor(hook);
+
+    it('TINY-8778: # triggers heading match when block tag matches forced_root_block', () => {
```

Comment:
This does not relate to the new lookup feature, so it should not be using `TINY-8778` as the Jira reference. It should just be using `TBA` since we don't know what it applies to.

- [ ] Fix incorrect Jira number references in tests